### PR TITLE
Mailet 128 Test and rewrite ToRepository

### PR DESCRIPTION
**Pull request by rouazana**

### CODE REVIEW

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToRepository.java [hunk 1]
```diff
@@ -59,31 +47,28 @@ public void setStore(MailRepositoryStore mailStore) {
         this.mailStore = mailStore;
     }
 
-    /**
-     * Initialize the mailet, loading configuration information.
-     */
```
Comment (aduprat) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToRepository.java]: @Override

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToRepository.java [hunk 2]
```diff
@@ -59,31 +47,28 @@ public void setStore(MailRepositoryStore mailStore) {
...
         }
-
     }
 
-    /**
-     * Store a mail in a particular repository.
-     * 
-     * @param mail
-     *            the mail to process
-     */
```
Comment (aduprat) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToRepository.java]: @Override

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToRepository.java [hunk 3]
```diff
@@ -93,11 +78,6 @@ public void service(Mail mail) throws javax.mail.MessagingException {
         }
     }
 
-    /**
-     * Return a string describing this mailet.
-     * 
-     * @return a string describing this mailet
-     */
```
Comment (aduprat) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToRepository.java]: @Override

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToRepository.java (hunk 2 above)
Comment (rouazana) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToRepository.java]: ok

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToRepository.java (hunk 3 above)
Comment (rouazana) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToRepository.java]: ok

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToRepository.java (hunk 1 above)
Comment (rouazana) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToRepository.java]: ok

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToRepository.java [hunk 4]
```diff
@@ -59,31 +47,28 @@ public void setStore(MailRepositoryStore mailStore) {
...
     public void init() throws MessagingException {
         repositoryPath = getInitParameter("repositoryPath");
+        passThrough = getPassThroughParameter();
+        repository = selectRepository();
+    }
+
+    private boolean getPassThroughParameter() {
         try {
-            passThrough = Boolean.valueOf(getInitParameter("passThrough"));
+            return Boolean.valueOf(getInitParameter("passThrough"));
```
Comment (mbaechler) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToRepository.java]: `getInitParameter("passThrough", false)` ?

Comment (rouazana) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToRepository.java]: ok + fixed coverage of the test

**Comment by mbaechler** [comment]
:+1:

**Comment by aduprat** [comment]
:+1:

**Pull request merged**